### PR TITLE
fix(button)!: remove BaseButton

### LINKED ISSUE #2616: Remove BaseButton

### PR DESCRIPTION
**Pull request by bennypowers**
Closes #2616

## What I did

1. remove basebutton

## Testing Instructions

1. check dp against prod demos

## Notes to Reviewers

1. see PR in rhds
2. @eyevana does this affect cp-elements
3. don't merge until 2.5.0 is release with all non-breaking feats in queue merged

### CODE REVIEW

**Comment by zeroedin** [comment]
> 2\. @eyevana does this affect cp-elements

@eyevana I added you as a reviewer as a ping for this question.

File: elements/pf-button/pf-button.ts [hunk 1]
```diff
@@ -182,22 +185,71 @@ export class PfButton extends BaseButton {
...
+              ?disabled="${this.disabled || this.#internals.formDisabled}">
+        <slot id="icon" part="icon" aria-hidden="true" name="icon">
+          <pf-icon
+              icon="${ifDefined(this.icon)}"
+              set="${ifDefined(this.iconSet)}"
+              ?hidden="${!this.icon}"></pf-icon>
+          <pf-spinner
+              ?hidden="${!this.loading}"
+              size="md"
+              aria-label="${this.getAttribute('loading-label') ?? 'loading'}"></pf-spinner>
```
Comment (zeroedin) [inline elements/pf-button/pf-button.ts:233]: Not related to this PR specifically but while I'm reviewing...   

If a icon is given along with the loading being set 
```
<pf-button icon="link" id="loading-logs">Resume loading logs</pf-button>
```
You end up with stacked icon and spinner.  Should `pfe-icon` be hidden if loading?  I couldn't find anything specific in PF that mentioned this combo.

```Suggestion
          <pf-icon
              icon="${ifDefined(this.icon)}"
              set="${ifDefined(this.iconSet)}"
              ?hidden="${!this.icon} || ${this.loading}"></pf-icon>
          <pf-spinner
              ?hidden="${!this.loading}"
              size="md"
              aria-label="${this.getAttribute('loading-label') ?? 'loading'}"></pf-spinner>
```

File: .changeset/remove-basebutton.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (zeroedin) [inline .changeset/remove-basebutton.md:1]:
```Suggestion
--- 
```
Ignore suggestion, blocking comment for approval.

**Don't merge until non-breaking changes are released in 2.5.0**

**Review by zeroedin — APPROVED** [review]
Given the non blocking suggestion otherwise, LGTM

**Comment by eyevana** [comment]
@bennypowers @zeroedin cp-elements isn't using any of the Base* classes, so no dependency from that end.

**Review by eyevana — APPROVED** [review]

File: elements/pf-button/pf-button.ts (hunk 1 above)
Comment (bennypowers) [reply elements/pf-button/pf-button.ts:233]: sounds reasonable

**Pull request merged**